### DISCUSSION TITLE
What is the replacement of rapidjson in cocos2dx v3.17.2

### DISCUSSION THREAD

**Discussion by nirobhasanbd** [axmolengine/axmol#917] [Q&A]
What is the replacement of rapidjson in cocos2dx v3.17.2. I am upgrading my cocos2dx v2.17.2 project to axmol.

I need help.

Thank you.

**Comment by rh101** [comment]
Can you please clarify what you mean? Is there a reason to not use rapidjson?

**Reply by nirobhasanbd** [reply]
@rh101
My existing project is in coco2dx v3.17.2. I used rapidjson to parse a json file. Now I want to upgrade my project to axmol v1.0.0. But I see the rapidjson is no longer exist in axmol v1.0.0. So I need to know what is the replacement of rapidjson in axmol.

**Reply by rh101** [reply]
I'm not sure why you think it no longer exists. Did you actually check?

![image](https://user-images.githubusercontent.com/8603230/194988656-33930787-3480-4cd7-9375-335031d83402.png)

Also, regardless of whether it is included as part of the engine or not, you can always add it or any other library to your game project.  It's straightforward to add a library with a few lines in your `CMakeLists.txt` file.

**Reply by johnspeny** [reply]
@rh101 , i find it hard to add other libraries to my game projects since some libraries are structured differently. I may need some tips on how to add them on my axmol projects.

**Reply by rh101** [reply]
If a library does not come with a `CMakeLists.txt`, then simply create one yourself.  Review whatever build files the library does use and port the required elements across to the `CMakeLists.txt`.  It's not difficult, but it is a little time consuming.  

This engine and thousands of other projects out there use CMake, so use them as a reference on how things are done, and if you need help related to CMake, then the official documentation site is the place to start: https://cmake.org/cmake/help/latest/index.html

How to use CMake is outside the scope of this discussion forum, and there are many other sites that cater for help with that.
